benchmarks/timeseries/ringbuffer_memusage.py

```diff
@@ -0,0 +1,91 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""Memory allocation benchmark for the ringbuffer."""
+
+from __future__ import annotations
+
+import argparse
+import tracemalloc
+from datetime import datetime, timedelta
+
+import numpy as np
+
+from frequenz.sdk.timeseries import Sample
+from frequenz.sdk.timeseries._ringbuffer import OrderedRingBuffer
+
+FIVE_MINUTES = timedelta(minutes=5)
+
+RINGBUFFER_LENGTH = 4_000
+# Number of iterations to run the benchmark
+ITERATIONS = 100
+
+
+def parse_args() -> tuple[int, int, int]:
+    """Parse command line arguments."""
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--size",
+        type=int,
+        default=RINGBUFFER_LENGTH,
+        help="Size of the ringbuffer to memory benchmark",
+    )
+    parser.add_argument(
+        "--iterations",
+        type=int,
+        default=ITERATIONS,
+        help="Number of iterations to run the benchmark",
+    )
+    parser.add_argument(
+        "--gap-size",
+        type=int,
+        default=0,
+        help="Number of samples to skip between each update",
+    )
+    args = parser.parse_args()
+    return args.size, args.iterations, args.gap_size
+
+
+def main(ringbuffer_len: int, iterations: int, gap_size: int) -> None:
+    """Run the benchmark.
+
+    Args:
+        ringbuffer_len: Size of the ringbuffer to dump/load
+        iterations: Number of iterations to run the benchmark
+        gap_size: Number of samples to skip between each update
+
+    """
+    # Trace memory allocations
+    tracemalloc.start()
+
+    ringbuffer = OrderedRingBuffer(
+        np.arange(0, ringbuffer_len, dtype=np.float64), timedelta(minutes=5)
+    )
+
+    # Snapshot memory allocations after ringbuffer creation
+    snapshot_init = tracemalloc.take_snapshot()
+
+    print("size: ", ringbuffer_len)
+    print("iterations: ", iterations)
+    print("gap_size: ", gap_size)
```

Comment:
you might want to use f-strings for better readability instead

```suggestion
    print(f"size: {ringbuffer_len}")
    print(f"iterations: {iterations}")
    print(f"gap_size: {gap_size}")
```

Reply:
I don't really see how it's better to read here, but :shrug:

Reply:
well it's the new python way of formatting strings and we use it for string formatting in the SDK except on logging.
In this case there is only a variable logged per line but the three lines could be merged easily into a line using f-string and still be readable. Moreover if the label matches the variable name (not in this case), we can just log them as `print(f"{size=} {iterations=} {gap_size}")

Reply:
Well, that last part was new to me :) Learned something.